packages/frameworks/vue/src/menu/menu-content.tsx

```diff
@@ -2,11 +2,13 @@ import { defineComponent } from 'vue'
 import { ark, type HTMLArkProps } from '../factory'
 import { Presence, type PresenceProps } from '../presence'
 import { emits, props } from '../presence/presence.props'
+import type { Assign } from '../types'
 import { useMenuContext } from './menu-context'
+import type { ComponentWithProps } from '../utils'
 
-export type MenuContentProps = HTMLArkProps<'div'> & PresenceProps
+export interface MenuContentProps extends Assign<HTMLArkProps<'div'>, PresenceProps> {}
 
-export const MenuContent = defineComponent({
+export const MenuContent: ComponentWithProps<MenuContentProps> = defineComponent({
```

Comment:
```suggestion
export const MenuContent = defineComponent({
```
We don't use `ComponentWithProps<P>` lately

Reply:
yes, i think so. `ComponentWithProps<P>` has a little hack. I think we just need define props correctly with `defineComponent`, or use `PropType` to annotate complex props types.

Reply:
> We don't use `ComponentWithProps<P>` lately. We could reconsider this. I found some little differences in intellisense, but in any way, this works. What do you think @iamdin?
> 
> Edit: According to #1884, it should be better to avoid this approach. Please remove each occurrence of `ComponentWithProps`编辑：根据#1884，最好避免这种方法。请删除所有出现的 `ComponentWithProps`

Yep, I will refactor with the new feature [Function Signature](https://vuejs.org/api/general.html#function-signature) released in Vue3.3

Reply:
Wow, they simplified a case for generic types in components, nice.

Reply:
Finally,  just remove `ComponentWithProps`